### ISSUE TITLE
Workaround for interceptor activation in Glassfish

### ISSUE THREAD

**Issue by Pscheidl** [Pscheidl/FortEE#14]
Workaround for @Failsafe interceptor not being activated in Glassfish-based AS. https://java.net/jira/browse/GLASSFISH-21718 / https://github.com/payara/Payara/issues/1532.

**Comment by Pscheidl** [comment]
From the CDI specs:
Interceptors enabled using @Priority are called before interceptors enabled using beans.xml.

An interceptor is said to be enabled if it is enabled in at least one bean archive or is listed in the final list of interceptors for the application, as defined in AfterTypeDiscovery event.


http://docs.jboss.org/cdi/spec/1.2/cdi-spec.html

**Comment by Pscheidl** [comment]
Enabling the interceptor during startup in AfterTypeDiscovery phase does not work. The root cause is the interceptor not being loaded. 

public void observeAfterTypeDiscovery(@Observes AfterTypeDiscovery afterTypeDiscovery) {
        afterTypeDiscovery.getInterceptors().add(FailsafeInterceptor.class);
    }

**Comment by Pscheidl** [comment]
Setting bean-discovery-mode in `beans.xml` to `all` resolves the interceptor discovery problem.

```xml
<?xml version="1.0" encoding="UTF-8"?>
<beans xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
       xmlns="http://xmlns.jcp.org/xml/ns/javaee"
       xsi:schemaLocation="http://xmlns.jcp.org/xml/ns/javaee http://xmlns.jcp.org/xml/ns/javaee/beans_1_1.xsd"
       bean-discovery-mode="all">

</beans>
```

**Comment by Pscheidl** [comment]
Fixed in new release 0.3.1
